### PR TITLE
[13.0][FWPORT] base_tier_validation: can_review fixes and secuiry

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
forward port of: https://github.com/OCA/server-ux/pull/150.

@ForgeFlow

### CODE REVIEW

File: base_tier_validation/models/tier_validation.py [hunk 1]
```diff
@@ -56,21 +47,30 @@ def _compute_has_comment(self):
...
+        sequence.sort()
+        my_sequence = sequence[0]
+        if self.review_ids.filtered(
+            lambda r: r.status != "approved" and r.sequence < my_sequence
+        ):
+            # A review has not been approved
+            return False
+        return True
+
+    @api.multi
```
Comment (AdriaGForgeFlow) [inline base_tier_validation/models/tier_validation.py]: This needs to be removed

File: base_tier_validation/models/res_users.py [hunk 2]
```diff
@@ -16,23 +16,30 @@ def review_user_count(self):
...
-                        "pending_count": 0,
-                    }
-                docs = to_review_docs.get(review.model)
-                if (docs and record not in docs) or not docs:
-                    user_reviews[review.model]["pending_count"] += 1
-                review.can_review = True
-                to_review_docs.setdefault(review.model, []).append(record)
+            record = (
+                review.env[review.model]
+                .sudo(self.env.user)
```
Comment (AdriaGForgeFlow) [inline base_tier_validation/models/res_users.py]: Shouldn't this be `with_user(self.env.user)`?

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]
Some comments:

**Comment by LoisRForgeFlow** [comment]
@AdriaGForgeFlow Fixed! Thanks, can you review again?

**Review by AdriaGForgeFlow — APPROVED** [review]

**Review by etobella — APPROVED** [review]

**Review by AdriaGForgeFlow — APPROVED** [review]

**Pull request merged**